python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -1486,6 +1496,97 @@ def mock_mean_none(self, *args, **kwargs):
     monkeypatch.setattr(xpd.Series.mean, "_fsproxy_slow", pd_mean)
 
 
+def mock_mean_memory_error(self, *args, **kwargs):
+    raise MemoryError()
+
+
+def mock_mean_rmm_error(self, *args, **kwargs):
+    raise RMMError(1, "error")
+
+
+# Parameterize the test function
+@pytest.mark.parametrize(
+    "mock_mean", [mock_mean_memory_error, mock_mean_rmm_error]
+)
+def test_cudf_pandas_debug_oom_fallback(monkeypatch, mock_mean):
+    cudf_mean = cudf.Series.mean
+
+    with monkeypatch.context() as monkeycontext:
+        monkeypatch.setattr(xpd.Series.mean, "_fsproxy_fast", mock_mean)
+        monkeycontext.setenv("CUDF_PANDAS_FALLBACK_DEBUGGING", "True")
+        s = xpd.Series([1, 2])
+        with pytest.warns(
+            CudfPandasDebugOOMWarning, match="Out of Memory Error."
+        ):
+            assert s.mean() == 1.5
+
+    # Must explicitly undo the patch. Proxy dispatch doesn't work with monkeypatch contexts.
+    monkeypatch.setattr(xpd.Series.mean, "_fsproxy_fast", cudf_mean)
+
+
+def test_cudf_pandas_debug_not_implemented_fallback(monkeypatch):
```

Comment:
These tests are similarly structured, so I don't mind using `pytest.mark.parameterize`.

Reply:
If you could use `parameterize` that would be great!